src/components/organisms/SchedulePageModal/SchedulePageModal.utils.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  addDays,
+  compareAsc,
+  differenceInMinutes,
+  endOfDay,
+  format,
+  isWithinInterval,
+  startOfDay,
+} from "date-fns";
+
+import { Room } from "types/rooms";
+import { PersonalizedVenueEvent, VenueEvent } from "types/venues";
+import { MyPersonalizedSchedule } from "types/User";
+
+import { WithVenueId } from "utils/id";
+import { ONE_MINUTE_IN_SECONDS, ONE_SECOND_IN_MILLISECONDS } from "utils/time";
+import { isTruthy } from "utils/types";
+
+import { RoomWithEvents } from "components/molecules/Schedule/Schedule.types";
+
+export const isEventThisDay = (date: Date) => {
+  return (event: VenueEvent) => {
+    return isWithinInterval(date, {
+      start: startOfDay(event.start_utc_seconds * ONE_SECOND_IN_MILLISECONDS),
+      end: endOfDay(
+        (event.start_utc_seconds +
+          event.duration_minutes * ONE_MINUTE_IN_SECONDS) *
+          ONE_SECOND_IN_MILLISECONDS
+      ),
+    });
+  };
+};
+
+export const adjustEventTiming = (date: Date) => {
+  return (event: WithVenueId<VenueEvent>) => {
+    const dates = [
+      event.start_utc_seconds * ONE_SECOND_IN_MILLISECONDS,
+      (event.start_utc_seconds +
+        event.duration_minutes * ONE_MINUTE_IN_SECONDS) *
+        ONE_SECOND_IN_MILLISECONDS,
+      startOfDay(date),
+      endOfDay(date),
+    ].sort(compareAsc);
+
+    const startUtcMilliseconds =
+      dates[1] instanceof Date ? dates[1].getTime() : dates[1];
+
+    return {
+      ...event,
+      start_utc_seconds: startUtcMilliseconds / ONE_SECOND_IN_MILLISECONDS,
+      duration_minutes: differenceInMinutes(dates[2], dates[1]),
+    };
+  };
+};
+
+export const extendRoomsWithDaysEvents = (
+  rooms: Room[],
+  daysEvents: WithVenueId<PersonalizedVenueEvent>[]
+): RoomWithEvents[] => {
+  return rooms.map((room) => {
+    const events: WithVenueId<PersonalizedVenueEvent>[] = daysEvents.filter(
+      (event) => event?.room === room?.title
+    );
+    return { ...room, events };
+  });
+};
+
+export const bookmarkPersonalizedEvents = (
+  usersEvents: MyPersonalizedSchedule
+) => (event: WithVenueId<VenueEvent>): WithVenueId<PersonalizedVenueEvent> => ({
+  ...event,
+  isSaved: isTruthy(event.id && usersEvents[event.venueId]?.includes(event.id)),
+});
+
+export const scheduleDayBuilder = (
+  today: Date,
+  events: WithVenueId<VenueEvent>[],
+  rooms: Room[],
+  usersEvents: MyPersonalizedSchedule
+) => (dayIndex: number) => {
+  const day = addDays(today, dayIndex);
+
+  const daysEvents: WithVenueId<PersonalizedVenueEvent>[] = events
+    .filter(isEventThisDay(day))
+    .map(adjustEventTiming(day))
+    .map(bookmarkPersonalizedEvents(usersEvents));
+
+  const roomsWithEvents = extendRoomsWithDaysEvents(rooms, daysEvents);
+
+  return {
+    isToday: dayIndex === 0,
+    weekday: format(day, "E"),
+    dayStartUtcSeconds: Math.floor(day.getTime() / ONE_SECOND_IN_MILLISECONDS),
```

Comment:
https://date-fns.org/v2.20.1/docs/startOfDay ?

Reply:
seems that in this case I needed `getUnixTime`:  https://github.com/sparkletown/sparkle/pull/1302/commits/9fce282d69bf3236a3cad6381d0b990072e0c3c2

Reply:
Oh? Is `dayStartUtcSeconds` named poorly then? I don't think `getUnixTime` is going to give us the start of the day is it?

I would have expected `getUnixTime(startOfDay(day))` or similar would be the needed code here?

Reply:
probably I need to rename `day` to `dayStart` because I process only the beginnings of the days.

Reply:
renamed: https://github.com/sparkletown/sparkle/pull/1302/commits/b5f6424790abf0724aaae75f27e884345d8c59f5